lib/parser/provenance.cc

```diff
@@ -328,6 +360,45 @@ std::optional<ProvenanceRange> CookedSource::GetProvenanceRange(
   return {ProvenanceRange{first.start(), last.start() - first.start()}};
 }
 
+std::optional<CharBlock> CookedSource::GetCharBlockFromLineAndColumns(
+    int line, int column1, int column2) const {
+  // 2nd column is exclusive, meaning it is target column + 1.
+  auto provenanceStart{allSources_.GetFirstFileProvenance().value().start()};
+  if (auto sourceFile{allSources_.GetSourceFile(provenanceStart)}) {
+    if (auto firstOffset{
+            provenanceMap_.ReverseMap(sourceFile->GetLineStartOffset(line) +
+                provenanceStart.offset() + column1 - 1)}) {
+      if (auto secondOffset{
+              provenanceMap_.ReverseMap(sourceFile->GetLineStartOffset(line) +
+                  provenanceStart.offset() + column2 - 2)}) {
+        if (*secondOffset >= *firstOffset) {
+          // Returned 2nd column is also exclusive.
+          return CharBlock(
+              &data_[0] + *firstOffset, *secondOffset - *firstOffset + 1);
```

Comment:
I feel `&data[*firstOffset]` would be easier to read.

Reply:
Why?

Reply:
Well, ideally for an array x[] `x+offset` would be the most readable way. But since `data` is an std::string which only allows array access syntax, it is easier to read it as address of the char at `firstOffset` position rather than, address of first element + `firstOffest`. I suppose this is a subjective preference.

Reply:
Agree; thanks.